### ISSUE TITLE
[MonoVM] Profiled AOT generates "empty" shared libraries

### ISSUE THREAD

**Issue by grendello** [dotnet/runtime#61415]
Context: https://github.com/xamarin/xamarin-android/issues/3932

The DSOs contain no code, only data and that makes the startup/run performance hurt a bit (I observed times to load a DSO between 15 and 30ms on Android devices) because the DSO will be loaded first only to determine that there's no code in it and proceed to load the managed assembly. A Hello World MAUI app currently includes 120 DSOs, with roughly half of them empty.

I'm currently working on code which will prevent us from packaging such "empty" DSOs, but it would be great if they weren't generated at all :)

/cc @vargaz

**Comment by jonathanpeppers** [comment]
This might be related to: https://github.com/xamarin/xamarin-android/issues/3932

Some of these `.so` files are empty because it is Profiled AOT?

**Comment by grendello** [comment]
@jonathanpeppers yes, but that's the point exactly - we want to avoid packing empty DSOs since they degrade performance.

I've just finished the code to ignore them and the final tally is 102 out of 115 assemblies were empty for a Profile AOT-d MAUI Hello World app

**Comment by vargaz** [comment]
We can add an option to not generate the output, but most build systems like the build tools to have a consistent set of outputs, it makes writing build rules/dependencies etc. easier.

**Comment by lambdageek** [comment]
@vargaz Can we emit a length 0 file or a file with a completely empty `.text` if there's no "real" AOTd code in there?

**Comment by lambdageek** [comment]
@vargaz another idea: what if we emit a public symbol in `.data` if the assembly contains 0 real methods.  That's something that XA tooling can check for and skip bundling the shared library
